### PR TITLE
SQL math functions(round, abs, ceil, sqrt)

### PR DESCRIPTION
**Pull request by Macbeth98**
# Description

- This PR contains the implementation of Math functions such as `round, ceil, abs, sqrt` for the group by and non-group by SQL commands.
- The application of these functions to the column values is being done at `readAllRawRecords` function in `recordreader.go`.
- This way, these Math Evaluator functions can be applied when the values are read without the need to loop again.

# Features 
- **Implements Math Functions for both Group by and Non-Group by SQL Commands.**

# Testing
- Test SQL Commands
   - `select city, round(latitude, 2)`
   - `select city, round(latitude, 2) as lat_round, abs(longitude) as long_abs, sqrt(latency) as latency_sqrt`
   - `select city, ceil(sum(latitude)) as lat_sum from t group by city`
   - `select city, abs(sum(latitude)) as lat_sum from t group by city`
- Written test cases in `sqlparser_test.go` for the verifying of the parsing of SQL Math functions.

# Checklist:

- [X] I have self-reviewed this PR.
- [X] I have removed all print-debugging and commented-out code that should not be merged.
- [X] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [X] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

# Notice
- This is a PR on top of the PR #256. 
- I have chosen to do this way, as I worked on the addition of the SQL Round function, which is a part of the Math function, and the aliasing for aggregation and Round function columns.
- So, I wanted to make sure that aliasing will still be working for group and non-group commands. For that reason, I built this PR on top of the previous PR.   

# Reference Image
image

### CODE REVIEW

**Comment by Macbeth98** [comment]
@AndrewHess @nkunal 

Please review this PR and let me know if there need to be any further changes. 
Also, let me know if you want me to convert this into a Draft PR, till the previous one is merged.

File: pkg/segment/aggregations/segaggs.go [hunk 1]
```diff
@@ -168,7 +168,36 @@ RenamingLoop:
 	}
 
 	if colReq.RenameColumns != nil {
-		return errors.New("performColumnsRequest: processing ColumnsRequest.RenameColumns is not implemented")
+		// return errors.New("performColumnsRequest: processing ColumnsRequest.RenameColumns is not implemented")
+
```
Comment (AndrewHess) [inline pkg/segment/aggregations/segaggs.go:172]: Don't just comment it out, delete it

**Review by AndrewHess — COMMENTED** [review]
This looks great. Needs one quick fix and then I think it's ready to merge

File: pkg/segment/aggregations/segaggs.go (hunk 1 above)
Comment (Macbeth98) [reply pkg/segment/aggregations/segaggs.go]: Done, @AndrewHess

**Review by AndrewHess — APPROVED** [review]

**Pull request merged**